CMakeLists.txt

```diff
@@ -100,7 +100,9 @@ endif()
 
 # --- Linux Joystick API -------------------------------------------------------
 
-check_include_files(linux/joystick.h HAVE_LINUX_JOYSTICK_H)
+if(NOT CORE_SYSTEM_NAME STREQUAL android)
```

Comment:
if the platform doesn't provide `linux/joystick.h`, then it's not linux. this check seems pretty explicit. Don't we have multiple linux systems? I figured it was better to blacklist android that whitelist everything else.